File: MDS2-2025/frontend/views/vista-verdetalles.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/horizontal-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/button/src/vaadin-button.js';

@customElement('vista-verdetalles')
export class VistaVerdetalles extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout style="position:absolute;width: 100%; height: 100%;" id="vaadinVerticalLayout">
 <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout1" style="align-self: stretch;">
  <label id="label" style="flex-grow: 1; ">Nick:</label>
  <span id="nick" style="align-self: center; flex-grow: 1;">Nick</span>
 </vaadin-horizontal-layout>
 <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout" style="align-self: stretch;">
  <label id="label2" style="flex-grow: 1; ">Correo:</label>
  <span id="correo" style="align-self: center; flex-grow: 1;">Correo</span>
 </vaadin-horizontal-layout>
 <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout2" style="align-self: stretch;">
  <label id="label1" style="flex-grow: 1; ">Texto:</label>
  <span id="texto" style="align-self: center; flex-grow: 1;">Texto</span>
 </vaadin-horizontal-layout>
 <vaadin-button id="volver" tabindex="0">
   Volver 
 </vaadin-button>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: MDS2-2025/src/main/java/vistas/VistaVerdetalles.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;

@Tag("vista-verdetalles")
@JsModule("./views/vista-verdetalles.ts")
public class VistaVerdetalles extends LitTemplate {

    @Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;
	@Id("nick")
	private Span nick;
	@Id("correo")
	private Span correo;
	@Id("texto")
	private Span texto;
	@Id("volver")
	private Button volver;
	@Id("label")
	private Label label;
	@Id("vaadinHorizontalLayout")
	private HorizontalLayout vaadinHorizontalLayout;
	@Id("vaadinHorizontalLayout1")
	private HorizontalLayout vaadinHorizontalLayout1;
	@Id("vaadinHorizontalLayout2")
	private HorizontalLayout vaadinHorizontalLayout2;
	@Id("label1")
	private Label label1;
	@Id("label2")
	private Label label2;

	public VistaVerdetalles() {
        // You can initialise any data required for the connected UI components here.
    }

	public Element getVaadinVerticalLayout() {
		return vaadinVerticalLayout;
	}

	public void setVaadinVerticalLayout(Element vaadinVerticalLayout) {
		this.vaadinVerticalLayout = vaadinVerticalLayout;
	}

	public Span getNick() {
		return nick;
	}

	public void setNick(Span nick) {
		this.nick = nick;
	}

	public Span getCorreo() {
		return correo;
	}

	public void setCorreo(Span correo) {
		this.correo = correo;
	}

	public Span getTexto() {
		return texto;
	}

	public void setTexto(Span texto) {
		this.texto = texto;
	}

	public Button getVolver() {
		return volver;
	}

	public void setVolver(Button volver) {
		this.volver = volver;
	}

}